prov/rxm/src/rxm_ep.c

```diff
@@ -2056,6 +2058,24 @@ static void rxm_ep_sar_init(struct rxm_ep *rxm_ep)
 	}
 }
 
+static int
+rxm_ep_inject_pkt_alloc(struct rxm_ep *rxm_ep, struct rxm_pkt **inject_pkt,
+			uint8_t op, uint64_t comp_flags)
+{
+	*inject_pkt = calloc(1, rxm_ep->msg_info->tx_attr->inject_size +
```

Comment:
for possible future change: Could inject packets use alloca() instead of calloc(), or do we need to keep the inject_pkt around until the completion?

Reply:
`rxm_ep_inject_pkt_alloc` is used to pre-allocate buffer for non-threadsafe case. We can keep the inject_pkt and tinject_pkt as a filed of rxm_ep instead of pointer to the objects, but I want to keep RxM EP objects (that is referenced within critical paths) as small as possible.

Reply:
Yes, right. This is the same approach like in the verbs provider implemented by you, but we store the pointer to the buffer instead the entire buffer object.